feature-libs/quote/components/details/overview/quote-details-overview.component.ts

```diff
@@ -5,24 +5,181 @@
  */
 
 import { Component } from '@angular/core';
-import { Quote, QuoteActionType, QuoteFacade } from '@spartacus/quote/root';
-import { TranslationService } from '@spartacus/core';
-import { Card } from '@spartacus/storefront';
-import { Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
+import {
+  Quote,
+  QuoteState,
+  QuoteAction,
+  QuoteActionType,
+  QuoteDetailsReloadQueryEvent,
+  QuoteMetadata,
+  QuoteFacade,
+} from '@spartacus/quote/root';
+import { EventService, TranslationService } from '@spartacus/core';
+import {
+  Card,
+  FocusConfig,
+  ICON_TYPE,
+  SaveCardEvent,
+} from '@spartacus/storefront';
+import { combineLatest, Observable } from 'rxjs';
+import { map, take } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-quote-details-overview',
   templateUrl: './quote-details-overview.component.html',
 })
 export class QuoteDetailsOverviewComponent {
   quoteDetails$: Observable<Quote> = this.quoteFacade.getQuoteDetails();
+  iconTypes = ICON_TYPE;
+  saveMode = false;
 
   constructor(
     protected quoteFacade: QuoteFacade,
+    protected eventService: EventService,
     protected translationService: TranslationService
   ) {}
 
+  focusConfig: FocusConfig = {
+    trap: false,
+    block: true,
+    autofocus: 'input',
+    focusOnEscape: false,
+  };
+
+  protected defineEmptyQuoteMetaData(): QuoteMetadata {
+    return {
+      description: undefined,
+      expirationTime: undefined,
+      name: undefined,
+    };
+  }
+
+  protected defineQuoteMetaData(saveCardEvent: SaveCardEvent): QuoteMetadata {
+    const metaData: QuoteMetadata = this.defineEmptyQuoteMetaData();
+    const metaDataPropertyNames = Object.getOwnPropertyNames(metaData);
+    const saveCardEventPropertyNames =
+      Object.getOwnPropertyNames(saveCardEvent);
+
+    metaDataPropertyNames?.forEach((name) => {
+      const index = saveCardEventPropertyNames.findIndex((element) =>
+        element.includes(name + '_')
+      );
+      if (index >= 0) {
+        const propertyName = saveCardEventPropertyNames[index];
+        metaData[name] = saveCardEvent[propertyName];
+      }
+    });
+    return metaData;
+  }
+
+  editQuote(quote: Quote, saveCardEvent: SaveCardEvent) {
+    this.saveMode = false;
+    const metaData = this.defineQuoteMetaData(saveCardEvent);
+
+    this.quoteFacade
+      .editQuote(quote.code, metaData)
+      .pipe(take(1))
+      .subscribe(
+        // success
+        () => {
+          this.eventService.dispatch({}, QuoteDetailsReloadQueryEvent);
+        }
+      );
+  }
+
+  isSaveMode(): boolean {
+    return this.saveMode;
+  }
+
+  isEditable(quote: Quote): boolean {
+    return (
+      (quote.state &&
+        (quote.state === QuoteState.BUYER_DRAFT ||
+          quote.state === QuoteState.SELLER_DRAFT)) ||
+      quote.isEditable
+    );
```

Comment:
```suggestion
    return (
      quote.state === QuoteState.BUYER_DRAFT ||
      quote.state === QuoteState.SELLER_DRAFT ||
      quote.isEditable
    );
```
1) i think this can be a bit simplified, as we can compare to undefined savefly - see my suggestion
2) do we need the additional check for the status at all?  I believe a quote in status QuoteState.BUYER_DRAFT or QuoteState.SELLER_DRAFT will be marked as isEditable already in the action normalizer. 

so in the end you might just `return quote.isEditable;`